spec/features/app_group_management/update_app_group_spec.rb

```diff
@@ -6,35 +6,75 @@
   let(:admin) { create(:user) }
 
   before(:each) do
-    set_check_user_groups({ 'groups' => [] })
+    set_check_user_groups('groups' => [])
 
     @app_group_a = create(:app_group)
     create(:infrastructure, app_group: @app_group_a)
   end
 
   describe 'Edit metadata' do
     context 'As Superadmin' do
-      scenario 'User can edit the app group metadata', js: true do
-        set_check_user_groups({ 'groups' => ['barito-superadmin'] })
+      before :each do
+        set_check_user_groups('groups' => ['barito-superadmin'])
         create(:group, name: 'barito-superadmin')
         create(:app_group_role)
+
         login_as admin
+      end
 
+      scenario 'User can edit the app group metadata', js: true do
         visit root_path
         click_link @app_group_a.name
 
         expect(page).to have_css("input#app_group_name[value='#{@app_group_a.name}']")
 
-        fill_in "app_group_name", with: "new_#{@app_group_a.name}"
-        find("input#app_group_name").native.send_keys :enter
+        fill_in 'app_group_name', with: "new_#{@app_group_a.name}"
+        find('input#app_group_name').native.send_keys :enter
 
         expect(page).to have_css("input#app_group_name[value='new_#{@app_group_a.name}']")
       end
+
+      scenario 'User can edit log retention days', js: true do
+        visit root_path
+        click_link @app_group_a.name
+
+        fill_in 'app_group_log_retention_days', with: '100'
+        find('input#app_group_log_retention_days').native.send_keys :enter
+
+        expect(page).to have_css("input#app_group_log_retention_days[value='100']")
+      end
+
+      scenario 'User can edit max TPS', js: true do
+        visit root_path
+        click_link @app_group_a.name
+
+        fill_in 'app_group_infrastructure_options_max_tps', with: '32850'
+        find('input#app_group_infrastructure_options_max_tps').native.send_keys :enter
+
+        expect(page).to have_css("input#app_group_infrastructure_options_max_tps[value='32850']")
+      end
+
+      scenario 'Max TPS editing preserves other options', js: true do
+        visit root_path
+        click_link @app_group_a.name
+
+        infrastructure = @app_group_a.infrastructure
+        infrastructure.update(options: { max_tps: 100, preserve_me: 1 })
+
+        fill_in 'app_group_infrastructure_options_max_tps', with: '200'
+        find('input#app_group_infrastructure_options_max_tps').native.send_keys :enter
+
+        infrastructure.reload
+        expect(infrastructure.options).to include('preserve_me')
+      end
     end
 
     context 'As Authorized User based on Role' do
       scenario 'User with role "owner" or "admin" can edit app groups metadata', js: true do
```

Comment:
So with this condition, owner and admin can also edit other metadata field (log retention and max tps), right? if yes, I think we need to discuss more with @tarabaskara and @ekarisky. Should we allow owner and admin to edit log retention and max tps or it can be change only by superadmin?

Reply:
No, no. They can't. Only superadmin which is able to change those parameters.

Reply:
Check this [file](https://github.com/BaritoLog/BaritoMarket/blob/f789d574fe50353e47fc24d810bb29c6839077bb/app/policies/app_group_policy.rb#L14-L16) where the policy is defined. Only Barito superadmin who can update an app group.

Reply:
Done. Please check. @fadlinurhasan